src/libraries/Microsoft.PowerFx.Connectors/OpenApiExtensions.cs

```diff
@@ -431,26 +431,27 @@ internal static ConnectorType GetConnectorType(this ISwaggerParameter openApiPar
 
                         case "byte": // Base64 string
                         case "binary": // octet stream
-                            return new ConnectorType(schema, openApiParameter, FormulaType.Blob);
-
-                        case "enum":
-                            if (schema.Enum.All(e => e is OpenApiString))
-                            {
-                                OptionSet optionSet = new OptionSet("enum", schema.Enum.Select(e => new DName((e as OpenApiString).Value)).ToDictionary(k => k, e => e).ToImmutableDictionary());
-                                return new ConnectorType(schema, openApiParameter, optionSet.FormulaType);
-                            }
-                            else
-                            {
-                                return new ConnectorType(error: $"Unsupported enum type {schema.Enum.GetType().Name}");
-                            }
+                            return new ConnectorType(schema, openApiParameter, FormulaType.Blob);
+                    }
 
-                        default:
-                            return new ConnectorType(schema, openApiParameter, FormulaType.String);
+                    if (schema.Enum != null && (settings.Compatibility == ConnectorCompatibility.CdpCompatibility || schema.Format == "enum"))

```

Comment:
> (settings.Compatibility == ConnectorCompatibility.CdpCompatibility || schema.Format == "enum") [](http://example.com/codeflow?start=46&length=95)

similar, can you move this to extension method? 

The idea is that all the policy around what the settings mean is then captured in a central place.